pkg/decode/decode_protobuf.go

```diff
@@ -139,6 +142,67 @@ func RecordToMap(fr *flow.Record) config.GenericMap {
 	return out
 }
 
+func PacketToMap(pr *flow.PacketRecord) config.GenericMap {
+	out := config.GenericMap{}
+
+	if pr == nil {
+		return out
+	}
+
+	packet := gopacket.NewPacket(pr.Stream, layers.LayerTypeEthernet, gopacket.Lazy)
+	if ethLayer := packet.Layer(layers.LayerTypeEthernet); ethLayer != nil {
+		eth, _ := ethLayer.(*layers.Ethernet)
+		out["SrcMac"] = eth.SrcMAC.String()
+		out["DstMac"] = eth.DstMAC.String()
+	}
+
+	if tcpLayer := packet.Layer(layers.LayerTypeTCP); tcpLayer != nil {
+		tcp, _ := tcpLayer.(*layers.TCP)
+		out["SrcPort"] = tcp.SrcPort.String()
+		out["DstPort"] = tcp.DstPort.String()
+	} else if udpLayer := packet.Layer(layers.LayerTypeUDP); udpLayer != nil {
+		udp, _ := udpLayer.(*layers.UDP)
+		out["SrcPort"] = udp.SrcPort.String()
+		out["DstPort"] = udp.DstPort.String()
```

Comment:
you are missing `SCTP` protocol support

Reply:
sure I can add it !

Reply:
https://github.com/netobserv/netobserv-ebpf-agent/pull/364/commits/b39afec82fd6fcbb07ce465f7379e8441962c6c5